### DISCUSSION TITLE
NAS Folder listing - folder icons not being found

### DISCUSSION THREAD

**Discussion by ceejayemm** [rern/rAudio#1629] [Bug Reports]
I see that a lot of work has been happening  to improve the listings in various file modes.  This seems to have introduced a small issue (may be it was there earlier but I just haven't seen it before) as I don't use file mode very much.

It seems when you list files from the NAS area (might be elsewhere too but I only have files on my NAS), as the top level / artist folders are displayed rAudio tries to find image files in these folders which (obviously?) aren't there. This produces the following errors for example:

Mar 16 09:27:35 RAOffice nginx[358]: 2024/03/16 09:27:35 [error] 358#358: *38 open() "/srv/http/mnt/MPD/NAS/NAS_Music/thumb.jpg" failed (2: No such file or directory), client: 192.168.86.50, server: localhost, request: "GET /%2Fmnt%2FMPD%2FNAS%2FNAS_Music%2Fthumb.jpg?v=1710581255 HTTP/1.1", host: "raoffice", referrer: "http://raoffice/"
Mar 16 09:27:36 RAOffice nginx[358]: 2024/03/16 09:27:36 [error] 358#358: *38 open() "/srv/http/mnt/MPD/NAS/NAS_Music/ABBA/thumb.jpg" failed (2: No such file or directory), client: 192.168.86.50, server: localhost, request: "GET /%2Fmnt%2FMPD%2FNAS%2FNAS_Music%2FABBA%2Fthumb.jpg?v=1710581257 HTTP/1.1", host: "raoffice", referrer: "http://raoffice/"
Mar 16 09:27:36 RAOffice nginx[358]: 2024/03/16 09:27:36 [error] 358#358: *30 open() "/srv/http/mnt/MPD/NAS/NAS_Music/Bellowhead/thumb.jpg" failed (2: No such file or directory), client: 192.168.86.50, server: localhost, request: "GET /%2Fmnt%2FMPD%2FNAS%2FNAS_Music%2FBellowhead%2Fthumb.jpg?v=1710581257 HTTP/1.1", host: "raoffice", referrer: "http://raoffice/"

I have folders for each artist, with individual album folders within the artist folder as in:

RAOffice:/srv/http/mnt/MPD/NAS/NAS_Music  #  ls /srv/http/mnt/MPD/NAS/NAS_Music/ABBA      
 Gold  'Super Trouper'  'The Album'   Voyage

RAOffice:/srv/http/mnt/MPD/NAS/NAS_Music  #  ls /srv/http/mnt/MPD/NAS/NAS_Music/ABBA/Gold
'01 Dancing Queen.m4a'            '10 S.O.S.m4a'                              '19 Waterloo.m4a'
'02 Knowing Me, Knowing You.m4a'  '11 Chiquita.m4a'                            AlbumArtSmall.jpg
'03 Take A Chance On Me.m4a'      '12 Fernando.m4a'                            AlbumArt_{B5020207-474E-4720-5ECF-2026EEDEEB00}_Large.jpg
'04 Mamma Mia.m4a'                '13 Voulez Vous.m4a'                         AlbumArt_{B5020207-474E-4720-5ECF-2026EEDEEB00}_Small.jpg
'05 Lay All Your Love On Me.m4a'  '14 Gimme! Gimme! Gimme! (A Man After.m4a'   AlbumArt_{B5020207-474E-4720-9710-022EE73DEB00}_Large.jpg
'06 Super Trouper.m4a'            '15 Does Your Mother Know.m4a'               AlbumArt_{B5020207-474E-4720-9710-022EE73DEB00}_Small.jpg
'07 I Have A Dream.m4a'           '16 One Of Us.m4a'                           coverart.jpg
'08 The Winner Takes It All.m4a'  '17 The Name Of The Game.m4a'                folder.jpg
'09 Money, Money, Money.m4a'      '18 Thank You For The Music.m4a'             thumb.jpg

RAOffice:/srv/http/mnt/MPD/NAS/NAS_Music  #  ls /srv/http/mnt/MPD/NAS/NAS_Music/Bellowhead
Broadside  Hedonism

RAOffice:/srv/http/mnt/MPD/NAS/NAS_Music  #  ls /srv/http/mnt/MPD/NAS/NAS_Music/Bellowhead/Broadside
'01 Byker Hill.m4a'              '07 Thousands Or More.m4a'                      AlbumArtSmall.jpg
'02 Old Dun Cow.m4a'             '08 Dockside Rant_Sailing On The Tide.m4a'      AlbumArt_{B5020207-474E-4720-29E7-443F889CBC00}_Large.jpg
'03 Roll The Woodpile Down.m4a'  "09 The Wife Of Usher's Well.m4a"               AlbumArt_{B5020207-474E-4720-29E7-443F889CBC00}_Small.jpg
'04 10,000 Miles Away.m4a'       '10 What'\''s The Life Of A Man (Any Mor.m4a'   coverart.jpg
'05 Betsy Baker.m4a'             '11 Lillibulero.m4a'                            folder.jpg
'06 Black Beetle Pies.m4a'       '12 Go My Way.m4a'                              thumb.jpg

Is this an issue or me being pedantic ?

Chris

**Comment by rern** [comment]
Hi Chris,

It's a lot faster to let the browser tries loading `thumb.jpg` and fallback to default if it's missing. Otherwise rAudio has to look for the files in every directory which takes seconds to response for NAS with a lot of directories.

**Comment by ceejayemm** [comment]
Thanks for the explanation, it makes sense if not very pretty :-)

Chris
